### PR TITLE
Storyshots - Replace require_context.js with babel-plugin-require-context-hook

### PR DESCRIPTION
**Pull request by igor-dv**
Issue: #2894, #3406

## What I did

Moved webpack's `require.context()` polyfilling out of the storyshots implementation to the babel. It removes the complexity of manual transpyling `config.js`, but just **requiring** it.

### CODE REVIEW

File: MIGRATION.md [hunk 1]
```diff
@@ -46,6 +45,7 @@ The `@storybook/react-native` had built-in addons (`addon-actions` and `addon-li
 
 1. `imageSnapshot` test function was extracted from `addon-storyshots` and moved to a new package - `addon-storyshots-puppeteer` that now will be dependant on puppeteer
 2. `getSnapshotFileName` export was replaced with the `Stories2SnapsConverter` class that now can be overridden for a custom implementation of the snapshot-name generation
+3. Storybook that was configured with Webpack's `require.context()` feature will need to add a babel plugin to polyfill this functionality. A possible plugin might be [babel-plugin-require-context-hook](https://github.com/smrq/babel-plugin-require-context-hook)
```
Comment (tmeasday) [inline MIGRATION.md]: Is this advice for migrating users? Perhaps something more definite would be good here?

File: addons/storyshots/storyshots-core/README.md [hunk 2]
```diff
@@ -38,6 +38,53 @@ If you aren't familiar with Jest, here are some resources:
...
+  "plugins": ["..."],
+  "env": {
+    "test": {
+      "plugins": ["require-context-hook"]
+    }
+  }
+}
+```
+
+Make sure that it is added under the needed environment,
```
Comment (tmeasday) [inline addons/storyshots/storyshots-core/README.md]: "Make sure not to include this babel plugin in the config environment that applies to webpack" -- might be a bit clearer?

**Review by tmeasday — COMMENTED** [review]
Looking forward to getting rid of that crazy code!

A couple small README comments

File: MIGRATION.md (hunk 1 above)
Comment (igor-dv) [reply MIGRATION.md]: 🤔 I can link to the README explanation about this. WDYT?

File: addons/storyshots/storyshots-core/src/frameworks/config-loader.js [hunk 3]
```diff
@@ -1,43 +0,0 @@
-import fs from 'fs';
-import path from 'path';
-import { getBabelConfig } from '@storybook/core/server';
```
Comment (Hypnosphi) [inline addons/storyshots/storyshots-core/src/frameworks/config-loader.js]: This export can probably be removed from `@storybook/core` now, as well as `@storybook/<framework>/options.js` endpoints

Comment (igor-dv) [reply addons/storyshots/storyshots-core/src/frameworks/config-loader.js]: Will remove

**Review by Hypnosphi — APPROVED** [review]

**Pull request merged**